platform/security/core/security-core-impl/src/main/java/ddf/security/ws/policy/impl/PolicyWSDLGetInterceptor.java

```diff
@@ -102,6 +121,33 @@ public void handleMessage(Message message) throws Fault {
         }
     }
 
+    private Document configureAddress(Document doc) {
+        Element de = doc.getDocumentElement();
+        XPath xPath = XPathFactory.newInstance().newXPath();
+        String xPathString = String
+                .format("//@*[starts-with(.,\"%s\")] | //text()[starts-with(.,\"%s\")]",
+                        DEFAULT_ADDRESS, DEFAULT_ADDRESS);
+        NodeList nodes;
+        try {
+            nodes = (NodeList) xPath.evaluate(xPathString, de, XPathConstants.NODESET);
+        } catch (XPathExpressionException e) {
+            LOGGER.error("Could not evaluate the xPath expression: {}", xPathString, e);
+            return doc;
+        }
+        for (int i = 0; i < nodes.getLength(); i++) {
+            Node node = nodes.item(i);
+            node.setNodeValue(node.getNodeValue().replaceAll(DEFAULT_ADDRESS, getFullAddress()));
```

Comment:
You should call `getFullAddress()` outside this loop. And you could also, I think, optimize even further by checking `getFullAddress().equals(DEFAULT_ADDRESS)` at the top of the method and returning the unmodified doc.